app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -1918,11 +1918,6 @@ private void searchFiles() {
       if (ContextCompat.checkSelfPermission(this,
           Manifest.permission.READ_EXTERNAL_STORAGE)
           != PackageManager.PERMISSION_GRANTED) {
-        if (ActivityCompat.shouldShowRequestPermissionRationale(this,
```

Comment:
Since you have removed this `if` statement you can simply the `if` statements. So the entire method will reduce to
```java
  private void searchFiles() {
    if (Build.VERSION.SDK_INT >= VERSION_CODES.M && ContextCompat.checkSelfPermission(this,
        Manifest.permission.READ_EXTERNAL_STORAGE)
        != PackageManager.PERMISSION_GRANTED) {
      ActivityCompat.requestPermissions(this,
          new String[] { Manifest.permission.READ_EXTERNAL_STORAGE },
          REQUEST_READ_STORAGE_PERMISSION);
    } else {
      fileSearch.scan(sharedPreferenceUtil.getPrefStorage());
    }
  }
```